### PR TITLE
Auto-generate OSE inventory file

### PR DESCRIPTION
**Pull request by JayKayy**
#### What does this PR do?

This role is used to generate a valid inventory file for an Openshift Enterprise install
#### How should this be manually tested?

Simply use the inventory file in the `openshift_install/test` dir(change and add to it as you wish) and run it with the playbook. Then check the inventory file in the dir you ran it from for inventory_<env_id> for your ose installer inventory file.
[jkwiatko@fedora23 openshift_install]$ ansible-playbook -i test/ose-provision openshift_install.yaml 
#### Is there a relevant Issue open for this?

N/A
#### Who would you like to review this?

/cc @etsauer  @sabre1041

### CODE REVIEW

File: rhc-ose-ansible/integration.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+- hosts: localhost
+  vars_prompt:
```
Comment (etsauer) [inline rhc-ose-ansible/integration.yaml]: @JayKayy i don't think there's any reason to prompt for these. Just put them in your input inventory file, and show that they get output to the generated inventory file.

File: rhc-ose-ansible/roles/openshift_generate_inventory/meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#---
```
Comment (etsauer) [inline rhc-ose-ansible/roles/openshift_generate_inventory/meta/main.yml]: did you mean to leave these commented out?

**Comment by etsauer** [comment]
@JayKayy i think I would rename this role to just 'openshift-install'

File: rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2 [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Create an OSEv3 group that contains the master, nodes, etcd, and lb groups.
+# The lb group lets Ansible configure HAProxy as the load balancing solution.
+# Comment lb out if your load balancer is pre-configured.
+[OSEv3:children]
+masters
+nodes
+etcd
+lb
```
Comment (oybed) [inline rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2]: This one should probably have the same conditional as below - i.e. "{% if openshift_lbs is defined %}"

File: rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2 [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# Create an OSEv3 group that contains the master, nodes, etcd, and lb groups.
+# The lb group lets Ansible configure HAProxy as the load balancing solution.
+# Comment lb out if your load balancer is pre-configured.
+[OSEv3:children]
+masters
+nodes
+etcd
```
Comment (oybed) [inline rhc-ose-ansible/roles/openshift_generate_inventory/templates/inventory_template.j2]: This one should probably have the same conditional as below - i.e.: "{% if openshift_etcds is defined %}"

File: rhc-ose-ansible/integration.yaml (hunk 1 above)
Comment (JayKayy) [inline rhc-ose-ansible/integration.yaml]: @etsauer So this is where i'm slightly confused as i thought that the point was to not have an "inventory file"?

Comment (etsauer) [inline rhc-ose-ansible/integration.yaml]: @JayKayy the point is not to avoid having an inventory file. The point is that we will be feeding in an rhc-ose specific inventory file for our stuff... and as part of the provisioning process, we want to "produce" an inventory file that looks like what you'd expect for an openshift-ansible inventory. This is for the purposes of debugging/troubleshooting openshift-ansible stuff vs rhc-ose stuff.

Basically the rhc-ose inventory will have a bunch of openshift installer vars, but also a bunch of stuff that has nothing to do with the openshift installer. we want to separate those openshift installer vars out to a separate inventory.

**Comment by JayKayy** [comment]
Updated based on your feedback.
Changes:
- Removed role dependencies based on advice from ansible slack channel as how ansible handles role dependency is likely to change.
- Fixed a few bugs in the template (pointed out by @oybed) causing etcd and lb groups to appear regardless of actual group presence.
- added appropriate variables to ose-provision inventory file
- Changed role name to `openshift_install` ( @etsauer )
- Moved playbook to playbooks dir and added access to role directory

File: rhc-ose-ansible/inventory/ose-provision [hunk 5]
```diff
@@ -9,9 +8,15 @@ openstack_key_name=""
 #openshift_storage_disk_volume=/dev/vdb
 #openstack_flavor_name="m1.medium"
 #openstack_image_name="ose3_1-base"
-#openshift_openstack_master_storage_size: 10
-#openshift_openstack_node_storage_size: 10
-
+#openshift_openstack_master_storage_size=10
+#openshift_openstack_node_storage_size=10
+#env_id=
+inv_cluster_method=native
```
Comment (etsauer) [inline rhc-ose-ansible/inventory/ose-provision]: I think we want a more obvious prefix for these, like 'openshift-ansible_'

**Comment by etsauer** [comment]
Do we like the naming convention of the playbooks in rhc-ose-ansible/playbooks/registry? If so I think we should stick with that. I.e.

```
./rhc-ose-ansible/playbooks
  - ./openshift
    - ./byo.yml <-- this would be @JayKayy's current playbook in this PR
    - ./provision-openstack.yml <-- this would take the place of what's currently at ./rhc-ose-ansible/ose-provision.yml
```

cc: @oybed @sabre1041

File: rhc-ose-ansible/inventory/ose-provision (hunk 5 above)
Comment (JayKayy) [inline rhc-ose-ansible/inventory/ose-provision]: Sure agreed. I'm hesitant to go with `openshift_ansible...` as we don't want our vars to overload/conflict with variables they set or we may run into compatibility issues with some of their roles/playbooks

Comment (etsauer) [inline rhc-ose-ansible/inventory/ose-provision]: cc: @detiber, want to get your opinion on this. We're basically using this playbook & role as a "decoupler" from our playbooks and the openshift-ansible stuff.

Comment (detiber) [inline rhc-ose-ansible/inventory/ose-provision]: We won't use the openshift_ansible prefix for variables, but that also leads to very long names...  you could use rhc_ose_ as a prefix for a bit shorter name that will not conflict.

**Comment by detiber** [comment]
If these playbooks will only be used against the OpenStack instances being created, I would suggest moving away from generating an inventory file and rather taking advantage of a dynamic inventory file and mapping groups based on instance tags to the byo groups.

An example of how to invoke the playbook and map variables is here: https://github.com/2015-Middleware-Keynote/demo-ansible/blob/master/playbooks/openshift_setup.yml#L56-L98

An example of the group mapping can be seen here: https://github.com/2015-Middleware-Keynote/demo-ansible/blob/master/playbooks/group_setup.yml

It can also be done through a static inventory file like this:

```
---
[OSEv3:children]
masters
nodes
etcd

[masters:children]
tag_openshift-demo-jdetiber-host-type_master

[etcd:children]
tag_openshift-demo-jdetiber-host-type_master

[nodes:children]
tag_openshift-demo-jdetiber-host-type_master
tag_openshift-demo-jdetiber-host-type_node


[tag_openshift-demo-jdetiber-host-type_master]

[tag_openshift-demo-jdetiber-host-type_node]
```

In this case, you would have both the dynamic inventory script and the static inventory file in the same directory and would use the common directory as the inventory.

File: rhc-ose-ansible/inventory/ose-provision [hunk 6]
```diff
@@ -2,13 +2,14 @@
...
+#env_id=
 
 # The following variables can be used to override the defaults
 #openshift_master_count=1
 #openshift_node_count=2
 #openshift_storage_disk_volume=/dev/vdb
 #openstack_flavor_name="m1.medium"
-#openstack_image_name="ose3_1-base"
+openstack_image_name="_OS1_rhel-guest-image-7.2-20151102.0.x86_64.qcow2"
 #openshift_openstack_master_storage_size: 10
```
Comment (JayKayy) [inline rhc-ose-ansible/inventory/ose-provision:13]: these should probably be changed to 
`openshift_openstack_master_storage_size=10`

**Comment by etsauer** [comment]
Gonna close this PR, as it's no longer relevant.
